node/service/src/lib.rs

```diff
@@ -1361,3 +1369,63 @@ pub fn build_full(
 	#[cfg(not(feature = "polkadot-native"))]
 	Err(Error::NoRuntime)
 }
+
+struct RevertConsensus {
+	blocks: BlockNumber,
+	backend: Arc<FullBackend>,
+}
+
+impl ExecuteWithClient for RevertConsensus {
+	type Output = sp_blockchain::Result<()>;
+
+	fn execute_with_client<Client, Api, Backend>(self, client: Arc<Client>) -> Self::Output
+	where
+		<Api as sp_api::ApiExt<Block>>::StateBackend: sp_api::StateBackend<BlakeTwo256>,
+		Backend: sc_client_api::Backend<Block> + 'static,
+		Backend::State: sp_api::StateBackend<BlakeTwo256>,
+		Api: polkadot_client::RuntimeApiCollection<StateBackend = Backend::State>,
+		Client: AbstractClient<Block, Backend, Api = Api> + 'static,
+	{
+		babe::revert(client.clone(), self.backend, self.blocks)?;
+		grandpa::revert(client, self.blocks)?;
+		Ok(())
+	}
+}
+
+/// Reverts the node state down to at most the last finalized block.
+#[cfg(feature = "full-node")]
+pub fn revert(
+	client: Arc<Client>,
+	backend: Arc<FullBackend>,
+	blocks: BlockNumber,
+	config: Configuration,
+) -> sp_blockchain::Result<()> {
```

Comment:
Does this need to be `sp_blockchain::Result`? It's one of the few places I did not cleanup the stringly typed error messages.

Reply:
Because at the moment it was only used by a caller returning an `sp_blockchain::Result<()>`, the choice was taken just for convenience. But I see the point and  from a design perspective makes sense to change it to a `Result` inline with all the other functions here.